candle-examples/examples/llama/main.rs

```diff
@@ -97,7 +100,13 @@ fn main() -> Result<()> {
     };
 
     let device = candle_examples::device(args.cpu)?;
-    let dtype = if args.use_f32 { DType::F32 } else { DType::F16 };
+    let dtype = match args.dtype.as_deref() {
+        Some("f16") => DType::F16,
+        Some("bf16") => DType::BF16,
+        Some("f32") => DType::F32,
+        Some(dtype) => panic!("Unsupported dtype {dtype}"),
```

Comment:
`bail!` rather than `panic!`? Here it would be really close but if this code was copied and pasted to a library function `bail!` would sound much better.